protecode/util.py

```diff
@@ -310,3 +315,18 @@ def filter_func(other_dict: typing.Dict[str, str]):
         return result.product_id()
     else:
         return None
+
+
+def image_ref_with_digest_if_possible(
+    image_reference: str | oci.model.OciImageReference,
+    digest: cm.DigestSpec,
+) -> str:
+    if not (digest and digest.value):
+        return image_reference
+
+    image_reference = oci.model.OciImageReference.to_image_ref(image_reference=image_reference)
+
+    if image_reference.has_digest_tag:
+        return image_reference.original_image_reference
+
+    return image_reference.ref_without_tag + '@sha256:' + digest.value
```

Comment:
stylistic nit: preferably use f-str?

as an alternative idea - considering replacing the tag is a somewhat common problem: how about we add a method like `replace_tag(tag: str)` (or `with_tag(tag: str)`) to OciImageReferences as a convenience-shortcut?

as another idea for convenience, we might add `as_oci_tag` to cm.DigestSpec

In both cases, it is perfectly fine to not add those convenience methods for me.